File: 2023/tests/day16.test.js
const fs = require('fs');
const { part1, part2 } = require('../src/day16');

const example1 = `
.|...\\....
|.-.\\.....
.....|-...
........|.
..........
.........\\
..../.\\\\..
.-.-/..|..
.|....-|.\\
..//.|....
`.trim();

const input = fs.readFileSync('inputs/input16.txt', 'utf8');

describe('part 1', () => {
  test('example', () => {
    expect(part1(example1)).toBe(46);
  });

  test('solution', () => {
    expect(part1(input)).toBe(7496);
  });
});

describe('part 2', () => {
  test('example', () => {
    expect(part2(example1)).toBe(51);
  });

  test('solution', () => {
    expect(part2(input)).toBe(7932);
  });
});

File: 2020/src/test/kotlin/Day16Test.kt
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Nested
import org.junit.jupiter.api.Test
import kotlin.test.assertEquals

@DisplayName("Day 16")
class Day16Test {
    private val example1 = readText("day16_example1.txt")
    private val example2 = readText("day16_example2.txt")
    private val input = readText("day16.txt")
    private val day = Day16()

    @Nested
    @DisplayName("Part 1")
    inner class Part1 {
        @Test
        fun `Matches example 1`() {
            val answer = day.part1(example1)
            assertEquals(71, answer)
        }

        @Test
        fun `Matches solution`() {
            val answer = day.part1(input)
            assertEquals(27911, answer)
        }
    }

    @Nested
    @DisplayName("Part 2")
    inner class Part2 {
        @Test
        fun `Matches solution`() {
            val answer = day.part2(input)
            assertEquals(737_176_602_479, answer)
        }
    }
}
